aws_requests_auth/aws_auth.py

```diff
@@ -95,7 +100,7 @@ def __call__(self, r):
 
         # Create payload hash (hash of the request body content). For GET
         # requests, the payload is an empty string ('').
-        body = r.body if r.body else ''
+        body = r.body if r.body else bytes()
         payload_hash = hashlib.sha256(body).hexdigest()
```

Comment:
python2 vs pytnon3 string compability is a work in progress for me.  Could you explain why we run `.encode('utf-8')` on `hashlib.sha256(canonical_request.encode('utf-8')).hexdigest()` but not on `payload_hash = hashlib.sha256(body).hexdigest()`?